### PR TITLE
Fix overflown modal content overlapping on mobile Safari

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?

For some reason, if you have content inside the `EXPERIMENTAL_Modal` that exceed the max height (which is supposed to be fine because we have an `overflow-auto` class), on mobile Safari that content is not properly laid out and it can overlap with each other (see the screenshot below). I've found [this StackOverflow](https://stackoverflow.com/questions/42292912/flexbox-max-height-in-ie-and-safari) question that describes this exact same problem, and the only answer posted resolved this issue.

#### What problem is this solving?

[Running workspace](https://billing--checkoutio.myvtex.com/cart/add?sku=289).

You can see the modal in the payment step of the Checkout, if you select a credit card and click the "ver opções de parcelamento" button, the modal should appear.

But you must test this using either an iPhone, or using the macOS Simulator.

#### How should this be manually tested?

No manual tests, I'm afraid.

#### Screenshots or example usage

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>image</td>
<td>image</td>
</tr>
</tbody>
</table>

gif

#### Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Review by estacioneto — APPROVED** [review]
Looks nice! Just waiting for @thayannevls review, but LGTM :)

**Review by thayannevls — APPROVED** [review]

**Pull request merged**
